pallene/translator.lua

```diff
@@ -50,12 +50,24 @@ end
 
 function Translator:add_whitespace(start_index, stop_index)
     assert(self.last_index <= start_index)
-    assert(start_index <= stop_index+1)
+    assert(start_index <= stop_index + 1)
     self:add_previous(start_index - 1)
 
     local region = self.input:sub(start_index, stop_index)
-    local partial = region:gsub("%S", " ")
-    table.insert(self.partials, partial)
+    local start_pos = region:find("[\n\r]%s*$")
```

Comment:
What do you think about using `[ \t]` instead of `%s`? I just noticed that this `%s` will also match newlines, which is not what we want.